example/advection/redefine_parthenon_defaults.cpp

```diff
@@ -34,11 +37,16 @@ void MeshBlock::ProblemGenerator(ParameterInput *pin) {
   Container<Real> &rc = real_containers.Get();
   CellVariable<Real> &q = rc.Get("advected");
 
+  Real x1size = pmy_mesh->mesh_size.x1max - pmy_mesh->mesh_size.x1min;
+  Real x2size = pmy_mesh->mesh_size.x2max - pmy_mesh->mesh_size.x2min;
+  Real x3size = pmy_mesh->mesh_size.x3max - pmy_mesh->mesh_size.x3min;
+
   for (int k = 0; k < ncells3; k++) {
     for (int j = 0; j < ncells2; j++) {
       for (int i = 0; i < ncells1; i++) {
-        Real rsq = std::pow(pcoord->x1v(i), 2) + std::pow(pcoord->x2v(j), 2);
-        q(k, j, i) = (rsq < 0.15 * 0.15 ? 1.0 : 0.0);
+        q(k, j, i) = 1e-6 * sin(pcoord->x1v(i) / x1size * TWO_PI) *
+                     cos(pcoord->x2v(j) / x2size * TWO_PI) *
+                     cos(pcoord->x3v(k) / x3size * TWO_PI);
```

Comment:
Wait. I stand corrected. This isn't a shock anymore. We should be seeing 1st-order convergence.

Reply:
Actually wait. No I see the issue. This thing has the wrong wavelength. It's not periodic in the `X1 + X2 + X3` direction. The wavelength should be `2*pi*sqrt(x1size*x1size + x2size*x2size + x3size*x3size)`. That's probably why you're not seeing convergence.

Reply:
Yes, thanks for pointing that out. This is fixed now.